build-system/common/esbuild-babel.js

```diff
@@ -16,14 +16,74 @@
 
 const babel = require('@babel/core');
 const crypto = require('crypto');
-const fs = require('fs');
+const fs = require('fs-extra');
 const path = require('path');
 
+/**
+ * Directory where the babel filecache lives.
+ */
+const CACHE_DIR = path.resolve(__dirname, '..', '..', '.babel-cache');
+
+/**
+ * Cache for storing transformed files on both memory and on disk.
+ */
+class BabelTransformCache {
```

Comment:
I think this should eventually live in its own file so it can be shared across the various Babel callers. Follow up PR is good.

Reply:
Agreed! Although I'm not sure if I'll have bandwidth to keep hacking on infra until early April